packages/react-codemirror/src/lang-cypher/signature-help.ts

```diff
@@ -0,0 +1,121 @@
+import { EditorState, StateField } from '@codemirror/state';
+import { showTooltip, Tooltip } from '@codemirror/view';
+import {
+  CallClauseContext,
+  findParent,
+  FunctionInvocationContext,
+  ParserRuleContext,
+  parserWrapper,
+  signatureHelp,
+  StatementsContext,
+} from '@neo4j-cypher/language-support';
+import { CypherConfig } from './lang-cypher';
+
+function isMethodNameOrExpressionName(parent: ParserRuleContext) {
+  return (
+    parent instanceof FunctionInvocationContext ||
+    parent instanceof CallClauseContext ||
+    parent instanceof StatementsContext
+  );
+}
+
+function isMethodName(parent: ParserRuleContext) {
+  return (
+    parent instanceof FunctionInvocationContext ||
+    parent instanceof CallClauseContext
+  );
+}
+
+function getSignatureHelpTooltip(
+  state: EditorState,
+  config: CypherConfig,
+): Tooltip[] {
+  let result: Tooltip[] = [];
+  const schema = config.schema;
+  const tokens = parserWrapper.parsingResult.tokens;
+  const lastToken = tokens.filter((token) => token.channel == 0).at(-2);
+
+  if (schema && config.signatureHelp && lastToken) {
+    const pos = state.selection.main.head;
+    const tree = parserWrapper.parsingResult;
+    const isOpenBracket = lastToken.text === '(';
+    const isSeparator = lastToken.text === ',';
```

Comment:
I was expecting this to work even when the `)` was present:

https://github.com/neo4j/cypher-language-support/assets/10564538/f277dcb6-4c65-4124-beb6-4becf2ab0b8b

Is it a known limitation, we can addressing it in a follow up PR if it's hard to implement

Reply:
I've added a prop to the editor `offset` to be able to place the cursor somewhere that's not the end and added tests for the case at hand.

What doesn't work is still if you go back in the query and want the signature help triggered